python/pylibcudf/pylibcudf/strings/findall.pyx

```diff
@@ -38,3 +38,35 @@ cpdef Column findall(Column input, RegexProgram pattern):
         )
 
     return Column.from_libcudf(move(c_result))
+
+
+cpdef Column find_re(Column input, RegexProgram pattern):
```

Comment:
Should this go in a new file, or should this file be renamed? It seems incorrect to have `find_re` defined in the `findall` file, since they're different algorithms.

Reply:
I was told to keep match these with the header files. So `find_re` and `findall` are both declared in `findall.hpp`. They are common only in that both use regex I suppose.

Reply:
I didn't realize `find_re` was in `findall.hpp`. Then yes, this is fine.

Reply:
(But it does raise the question of why `find_re` is in `findall.hpp` -- it doesn't "find all")

Reply:
IIUC they will be combined into a common file as per https://github.com/rapidsai/cudf/pull/16742/files#r1781466464

Reply:
The `find_re` has more in common with `findall` than just `find` based on the parameters and how they are used just in not what they return. 

Perhaps `find_re()` would be better in `contains.hpp` which has `contains_re()`
Since [`find.hpp`](https://github.com/rapidsai/cudf/blob/branch-24.12/cpp/include/cudf/strings/find.hpp) has `contains()` and `find()` (among other similar functions)
So [`contains.hpp`](https://github.com/rapidsai/cudf/blob/branch-24.12/cpp/include/cudf/strings/contains.hpp) would have `contains_re()` and `find_re()`.
I could do that in a separate PR since this one is already approved.